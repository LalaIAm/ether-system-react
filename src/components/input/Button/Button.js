import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  background-color: salmon;
  border: 1px solid indianred;
  border-radius: 3px;
  color: snow;
  padding: 9px 12px;
`;

export const Button = ({ onClick, children }) => (
  <StyledButton onClick={onClick}>{children}</StyledButton>
);

export const ButtonSizes = ['small', 'medium', 'large'];
export const ButtonModes = ['primary', 'secondary', 'warning', 'success', 'info'];

Button.propTypes = {
  /** Content to be rendered inside a Button */
  children: PropTypes.node,
  /** Mode of the button */
  mode: PropTypes.oneOf(ButtonModes),
  /** Function to run on click */
  onClick: PropTypes.func,
  /** Size of the Button */
  size: PropTypes.oneOf(ButtonSizes),
};

Button.defaultProps = {
  onClick: () => {},
  size: 'medium',
}
