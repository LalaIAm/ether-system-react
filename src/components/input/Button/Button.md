### Primary Button
```js
<Button>Button</Button>
```

### Secondary Button
```js
<Button mode="secondary">Button</Button>
```
