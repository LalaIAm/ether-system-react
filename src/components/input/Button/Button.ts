import React from 'react';

const Button = ({ onClick: handleClick, children }) = (
  <button onClick={handleClick}>{children}</button>
);
