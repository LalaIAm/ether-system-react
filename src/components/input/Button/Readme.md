```js { "props": { "children": "Hello" } }
<Button size="large">Push Me</Button>
```
