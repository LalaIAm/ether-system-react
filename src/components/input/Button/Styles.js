import styled, { css } from 'styled-components';

import colorVariables from '../../../base/color/colorVariables';
import fontSizes from '../../../base/typography/fontSizes';

export const buttonModeStyles = {
  primary: css`
    background-color: ${colorVariables.buttonPrimaryBackground};
    border: 1px solid ${colorVariables.buttonPrimaryBorder};
    color: ${colorVariables.buttonPrimaryText};
    ${fontSizes.size1}

    &:hover {
      background-color: ${colorVariables.buttonPrimaryBackgroundHover};
    }
  `,
  secondary: css`
    background-color: ${colorVariables.buttonSecondaryBackground};
    border: 1px solid ${colorVariables.buttonSecondaryBorder};
    color: ${colorVariables.buttonSecondaryText};
    ${fontSizes.size1}

    &:hover {
      background-color: ${colorVariables.buttonSecondaryBackgroundHover};
    }
  `,
};

export const StyledButton = styled.button`
  ${p => buttonModeStyles[p.mode]};
  border-radius: 3px;
  cursor: pointer;
  padding: 9px 12px;
`;
