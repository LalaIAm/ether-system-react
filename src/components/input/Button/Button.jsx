import PropTypes from 'prop-types';
import React from 'react';

import { StyledButton } from './Styles';

export const Button = ({ children, mode, onClick }) => (
  <StyledButton onClick={onClick} mode={mode}>{children}</StyledButton>
);

export const ButtonModes = ['primary', 'secondary', 'warning', 'success', 'info'];

Button.propTypes = {
  /** Content to be rendered inside a Button */
  children: PropTypes.node,
  /** Mode of the button */
  mode: PropTypes.oneOf(ButtonModes),
  /** Function to run on click */
  onClick: PropTypes.func,
};

Button.defaultProps = {
  onClick: () => {},
  mode: 'primary',
};
