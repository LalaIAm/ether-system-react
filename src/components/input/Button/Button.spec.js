import React from 'react';
import 'jest-styled-components';

import { Button } from './Button';

it('should render without issue', () => {
  const subject = shallow(<Button>Hello</Button>);
  expect(subject).toMatchSnapshot();
});
