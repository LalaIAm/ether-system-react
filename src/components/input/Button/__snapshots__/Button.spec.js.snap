// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component when nothing specified should render without issue 1`] = `
<button
  className="hWpIoz"
  mode="primary"
  onClick={[Function]}
  size="medium"
>
  Hello
</button>
`;
